src/main/java/uk/gov/homeoffice/digital/sas/timecard/validators/timeentry/TimeEntryValidator.java

```diff
@@ -30,6 +35,87 @@ public boolean isValid(Object value, ConstraintValidatorContext context) {
         timeEntry.getActualEndTime());
 
     session.setHibernateFlushMode(FlushMode.AUTO);
-    return timeEntryClashes.isEmpty();
+    if (!timeEntryClashes.isEmpty()) {
+      ClashingProperty clashingProperty = getClashingProperty(timeEntry, timeEntryClashes);
+      var payload = timeEntryClashes.stream().map(this::transformTimeEntry)
+          .collect(Collectors.toCollection(ArrayList::new));
+
+      HibernateConstraintValidatorContext hibernateContext =
+          context.unwrap(HibernateConstraintValidatorContext.class);
+
+      hibernateContext.disableDefaultConstraintViolation();
+      hibernateContext.withDynamicPayload(payload);
+      hibernateContext
+          .buildConstraintViolationWithTemplate(
+              "Time periods must not overlap with another time period")
+          .addPropertyNode(clashingProperty.toString())
+          .addConstraintViolation();
+      return false;
+    }
+    return true;
+  }
+
+  private JSONObject transformTimeEntry(TimeEntry timeEntry) {
+    var result = new JSONObject();
+    result.put("startTime", timeEntry.getActualStartTime());
+    result.put("endTime", timeEntry.getActualEndTime());
```

Comment:
should these not match the payload/resource property names. e.g. 'actual...'

Reply:
Joanna and I had a conversation about this when we wrote this. I think the decision was made because we thought there was a plan for these to be changed to remove the "actual" part in the near future, but happy to change them if it would be better to have them aligned and change them all at once (or if that isn't actually the plan)

Reply:
Ah ok yeah that should be fine, as i'm sure the UI is hooked up to use these properly anyway.